src/main/java/com/evch/rrm/CustomDequeue.java

```diff
@@ -0,0 +1,270 @@
+package com.evch.rrm;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.IntFunction;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+public class CustomDequeue<E> implements Deque<E> {
+    CustomList<E> dequeue = new CustomList<>();
+
+    public CustomDequeue() {
+    }
+
+    public CustomDequeue(Collection<? extends E> c) {
+        addAll(c);
+    }
+
+    @Override
+    public void addFirst(E e) {
+        dequeue.add(0, e);
+    }
+
+    @Override
+    public void addLast(E e) {
+        dequeue.add(size() - 1, e);
+    }
+
+    @Override
+    public boolean offerFirst(E e) {
+        dequeue.add(0, e);
+        return true;
+    }
+
+    @Override
+    public boolean offerLast(E e) {
+        dequeue.add(size() - 1, e);
+        return true;
+    }
+
+    @Override
+    public E removeFirst() {
+        if (size() == 0) {
+            throw new NoSuchElementException();
+        }
+        return dequeue.remove(0);
```

Comment:
indeed it is convenient, however it's O(N) TC
please adjust the class to have both removeFirst / removeLast to be O(1) operations.